### PR TITLE
Various cleanup

### PR DESCRIPTION
**Pull request by phayes**
See https://www.drupal.org/node/2913690

I've done a major cleanup to the 8.x branch, before committing, I'd like a review please.

To summarize the changes:

1. Fixed the drupal composer repository
2. Added a .gitignore
3. Fixed all coder issues, the code is now Drupal coding-standards compliant.
4. Fixed tests:
     A. syncs local directory to docroot/modules/program on every test run, not just on install
     B. Adds a run-tests.sh and composer test command to facilitate running tests easily
     C. Update tests so that they are now passing on drupal core 8.3, the current version of drupal
     D. Adds purge module as a dev dependancy (for testing)
     E. Adds instructions for testing into README.md

### CODE REVIEW

**Comment by phayes** [comment]
Working on fixing tests on travis now.

File: src/Zone.php [hunk 1]
```diff
@@ -180,15 +180,15 @@ public function listZones() {
     try {
 
       if ($cached = $this->cache->get($cid)) {
-       return $cached->data;
+        return $cached->data;
       }
```
Comment (varvashenya) [inline src/Zone.php]: np: extra line.
Edit: it looks like it's a standard within this repo.

**Comment by phayes** [comment]
Now passes Travis and ready for review.

File: run-tests.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# Copy the program into the drupal installation
+mkdir -p docroot/modules/program && rsync -a . docroot/modules/program  --exclude \".idea\" --exclude bin --exclude \".git\" --exclude \".gitignore\" --exclude docroot --exclude \"*.make\" --exclude \".travis.yml\" --exclude vendor && rm -fr modules/contrib themes/contrib
```
Comment (varvashenya) [inline run-tests.sh]: np: extra space before 1st `--exclude`

**Review by varvashenya — APPROVED** [review]
Maybe it's time to switch travis build from php 7 for something more modern. PHP 7.0.23 looks a little bit old.

**Pull request merged**